(fix): update snapshot tests (#257)
chalk-ai/docs: snapshot-tests.mdx

@pytest.fixture(scope="session")
def snapshots(client: ChalkClient):
    # The dataset name will be set to the latest commit on the branch
    main_snapshot = client.get_dataset(dataset_name="user_snapshot")
    main_snapshot_df = snapshot.to_pandas().set_index(User.id)

    branch_snapshot = client.offline_query(
        input={User.id: main_snapshot_df.index.to_list()},
        recompute_features=True,
        wait=True,

>         run_asynchrounously=True

Reviewer: ```suggestion
        run_asynchronously=True
```

Author: Thanks for catching haha :)

    )
    branch_snapshot_df = branch_snapshot.to_pandas().set_index(User.id)

    return active_snapshot_df, branch_snapshot_df
```

> These fixtures serve to output two Pandas DataFrames: the main snapshot (which we store in

Author: ```suggestion
These fixtures serve to output two Pandas DataFrames: the main_snapshot_df (which is stored in
```

> These fixtures serve to output two Pandas DataFrames: the main_snapshot_df (which is stored in

Author: ```suggestion
These fixtures serve to output two Pandas DataFrames: the `main_snapshot_df` (which is stored in
```

> a dataset called `user_snapshot`) and the branch snapshot (which is generated

Author: ```suggestion
a dataset called `user_snapshot`) and the `branch_snapshot_df` (which is generated
```

    assert (main_snapshot[User.age] - branch_snapshot[User.age]).max() <= 2
```

In addition to the tests, we'll create a script which will run when we merge pull requests—this

> script will update the `user_snapshot` dataset with the latest branch code.

Author: ```suggestion
script will update the `user_snapshot` dataset with the one generated with our latest code.
```

```python
import os
from chalk.client import ChalkClient

branch = os.environ["GITHUB_REF_NAME"]
client = ChalkeClient(branch=branch)

dataset = client.get_dataset(dataset_name="user_snapshot")

dataset.recompute_features(branch=branch, wait=True, run_asynchrounously=True)
```

## Writing the GitHub Action

We will write two GitHub actions: one will run when we create a pull requests, the other

> will run when we merge into our main branch.

Author: ```suggestion
will run when we merge code into our main branch.
```

          pytest -s ./tests
```

> The second GitHub Action runs our update script to update the `user_snapshot` dataset with our latest snapshot. In runs whenever we push code to the main branch.

Reviewer: ```suggestion
The second GitHub Action runs our update script to update the `user_snapshot` dataset with our latest snapshot. It runs whenever we push code to the main branch.
```
